### DISCUSSION TITLE
Support Controls Drawn on a existing rendering Canvas

### DISCUSSION THREAD

**Discussion by RChrisCoble** [dotnet/Microsoft.Maui.Graphics.Controls#32] [Ideas]
Was super excited to see this repo show up on my Twitter feed.  This is a brilliant idea that allows creation of controls that will run anywhere .Net can run, as Microsoft.Maui.Graphics doesn't have any specific ties to Maui. (ie:  It will also render with Blazor client and server).

For use cases on my end we need to draw traditional controls on top of a graphical canvas:  Button, Checkbox, Radio buttons, Edit box, ComboBox, DateTime Picker, Calendar Control.  Having these controls being drawn with the same underlying 2D API that is rendering to the canvas is a dream come true.  

So, the request from my end is, with Blazor client or Server, allow an existing canvas to be passed into the controls as their render target.

Also, in the context of .Net 6, allow Blazor Client/Server code to leverage Microsoft.Maui.Graphics and Microsoft.Maui.Graphics.Controls natively.

Keep up the fantastic work!

**Comment by RChrisCoble** [comment]
Matthew is busy adding Blazor support to SkiaSharp in preparation to use it in Microsoft.Maui.Graphics.  This means your controls should be able to be used in Blazor as well.  

https://github.com/mono/SkiaSharp/pull/1811

**Comment by jsuarezruiz** [comment]
Definitely there are many possibilities!. Already have done some tests in Linux using SkiaSharp:
<img width="797" alt="graphicscontrols-gtk" src="https://user-images.githubusercontent.com/6755973/134816784-c97fe2d0-8e50-4050-836c-2625545b2e24.png">
(work in progress)

And there are other ideas that would love to try.
